In [7]:
# ライブラリのインポート
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [8]:
# 元となるデータの取得
df_origin = pd.read_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-7/user-7-nonull.csv")
df_origin.head()

/tmp/ipython-input-8-3264174739.py:2: DtypeWarning: Columns (33,68) have mixed types. Specify dtype option on import or set low_memory=False.
  df_origin = pd.read_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-7/user-7-nonull.csv")


Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0          1230701762          NaN   
1           1                 7718          1230709480        Mouse   
2           2                18880          1230720642  Eye Tracker   
3           3                23441          1230725203        Mouse   
4           4                39413          1230741175        Mouse   

  Project name Export date Participant name Recording name Recording date  \
0          JD4  2025/07/18           user-7     Recording8     2025/07/18   
1          JD4  2025/07/18           user-7     Recording8     2025/07/18   
2          JD4  2025/07/18           user-7     Recording8     2025/07/18   
3          JD4  2025/07/18           user-7     Recording8     2025/07/18   
4          JD4  2025/07/18           user-7     Recording8     2025/07/18   

  Recording date UTC  ... Original Media height Eye movement type  \
0         2025/07/18  ...                   NaN               NaN   
1         2025/07/18  ...                   NaN               NaN   
2         2025/07/18  ...                   NaN      EyesNotFound   
3         2025/07/18  ...                   NaN      EyesNotFound   
4         2025/07/18  ...                   NaN      EyesNotFound   

   Eye movement event duration Eye movement type index Fixation point X  \
0                          NaN                     NaN              NaN   
1                          NaN                     NaN              NaN   
2                        204.0                     1.0              NaN   
3                        204.0                     1.0              NaN   
4                        204.0                     1.0              NaN   

  Fixation point Y  Fixation point X (MCSnorm)  Fixation point Y (MCSnorm)  \
0              NaN                         NaN                         NaN   
1              NaN                         NaN                         NaN   
2              NaN                         NaN                         NaN   
3              NaN                         NaN                         NaN   
4              NaN                         NaN                         NaN   

   Mouse position X  Mouse position Y  
0               NaN               NaN  
1            1834.0             926.0  
2               NaN               NaN  
3            1827.0             919.0  
4            1811.0             904.0  

[5 rows x 84 columns]

# カラムの整理

In [9]:
nunique_series = df_origin.nunique(dropna=False).to_string()
print(nunique_series)

Unnamed: 0                                    24236
Recording timestamp                           24223
Computer timestamp                            24223
Sensor                                            3
Project name                                      1
Export date                                       1
Participant name                                  1
Recording name                                    1
Recording date                                    1
Recording date UTC                                1
Recording start time                              1
Recording start time UTC                          1
Recording duration                                1
Timeline name                                     1
Recording Fixation filter name                    1
Recording software version                        1
Recording resolution height                       1
Recording resolution width                        1
Recording monitor latency                         1
Average cali

In [10]:
# 欠損値も含めてユニークな値の個数をカウント
nunique_series = df_origin.nunique(dropna=False)

# ユニークな値が1つのカラム名リスト
single_unique_cols = nunique_series[nunique_series == 1].index.tolist()

In [11]:
# カラム名とそのユニーク値を出力
for col in single_unique_cols:
    unique_val = df_origin[col].unique()
    print(f"{col}: {unique_val}")


Project name: ['JD4']
Export date: ['2025/07/18']
Participant name: ['user-7']
Recording name: ['Recording8']
Recording date: ['2025/07/18']
Recording date UTC: ['2025/07/18']
Recording start time: ['16:49:27.713']
Recording start time UTC: ['07:49:27.713']
Recording duration: [419125]
Timeline name: ['Timeline1']
Recording Fixation filter name: ['Tobii I-VT (Fixation)']
Recording software version: ['24.21.435']
Recording resolution height: [1080]
Recording resolution width: [1920]
Recording monitor latency: [10]
Average calibration accuracy (mm): [20.8]
Average calibration accuracy (degrees): [1.77]
Average calibration accuracy (pixels): [72]
Average validation accuracy (mm): [17.2]
Average validation precision SD (mm): [2.7]
Average validation precision RMS (mm): [2.5]
Average validation accuracy (degrees): [1.55]
Average validation precision SD (degrees): [0.24]
Average validation precision RMS (degrees): [0.23]
Average validation accuracy (pixels): [60]
Average validation precision

## カラムの要素が1つしかない変数の説明
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-7`|
|Recording name| 録画名|`Recording3`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:49:27.713`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:49:27.713`|
|Recording duration| 録画時間|`419125`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`20.8`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.77`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`72`pixels|
|Average validation accuracy | 平均バリデーション精度|`17.2`mm|
|Average validation precision SD | 平均バリデーション予測SD|`2.7`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`2.5`mm|
|Average validation accuracy |平均バリデーション精度| `1.55`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.24`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.23`degrees|
|Average validation accuracy |平均バリデーション精度| `60`pixels|
|Average validation precision SD |平均バリデーション予測SD| `9`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `9`pixels|

In [12]:
# 削除した新しいデータフレーム
df_cleaned = df_origin.drop(columns=single_unique_cols)
df_cleaned.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0           0                    0          1230701762          NaN   
1           1                 7718          1230709480        Mouse   
2           2                18880          1230720642  Eye Tracker   
3           3                23441          1230725203        Mouse   
4           4                39413          1230741175        Mouse   

   Eyetracker timestamp           Event Event value  Gaze point X  \
0                   NaN  RecordingStart         NaN           NaN   
1                   NaN             NaN         NaN           NaN   
2          1.246045e+09             NaN         NaN           NaN   
3                   NaN             NaN         NaN           NaN   
4                   NaN             NaN         NaN           NaN   

   Gaze point Y  Gaze point left X  ...  Original Media height  \
0           NaN                NaN  ...                    NaN   
1           NaN                NaN  ...                    NaN   
2           NaN                NaN  ...                    NaN   
3           NaN                NaN  ...                    NaN   
4           NaN                NaN  ...                    NaN   

   Eye movement type  Eye movement event duration  Eye movement type index  \
0                NaN                          NaN                      NaN   
1                NaN                          NaN                      NaN   
2       EyesNotFound                        204.0                      1.0   
3       EyesNotFound                        204.0                      1.0   
4       EyesNotFound                        204.0                      1.0   

   Fixation point X  Fixation point Y  Fixation point X (MCSnorm)  \
0               NaN               NaN                         NaN   
1               NaN               NaN                         NaN   
2               NaN               NaN                         NaN   
3               NaN               NaN                         NaN   
4               NaN               NaN                         NaN   

   Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
0                         NaN               NaN               NaN  
1                         NaN            1834.0             926.0  
2                         NaN               NaN               NaN  
3                         NaN            1827.0             919.0  
4                         NaN            1811.0             904.0  

[5 rows x 57 columns]

## ユニークな値を取り除いたデータフレーム

In [13]:
print(df_cleaned.nunique(dropna=False).to_string())

Unnamed: 0                             24236
Recording timestamp                    24223
Computer timestamp                     24223
Sensor                                     3
Eyetracker timestamp                   24008
Event                                      9
Event value                                9
Gaze point X                            1566
Gaze point Y                             924
Gaze point left X                       1552
Gaze point left Y                        953
Gaze point right X                      1560
Gaze point right Y                       864
Gaze direction left X                  10548
Gaze direction left Y                   9918
Gaze direction left Z                   4291
Gaze direction right X                 10803
Gaze direction right Y                  9808
Gaze direction right Z                  4600
Pupil diameter left                     1555
Pupil diameter right                    1496
Pupil diameter filtered                 1539
Validity l

## Sensor事にデータフレームを分割

In [14]:
# NaNを含む行
df_sensor_nan = df_cleaned[df_cleaned['Sensor'].isna()]

# "Mouse" の行
df_sensor_mouse = df_cleaned[df_cleaned['Sensor'] == "Mouse"]

# "Eye Tracker" の行
df_sensor_eye = df_cleaned[df_cleaned['Sensor'] == "Eye Tracker"]


### 問題の既知代わりの時間を取得する

In [15]:
df_sensor_nan

Unnamed: 0  Recording timestamp  Computer timestamp Sensor  \
0               0                    0          1230701762    NaN   
221           221              2849609          1233551371    NaN   
15745       15745            261610059          1492311821    NaN   
15753       15753            261712077          1492413839    NaN   
17137       17137            301178639          1531880401    NaN   
17145       17145            301274656          1531976418    NaN   
17161       17161            301628584          1532330346    NaN   
17162       17162            301628584          1532330346    NaN   
17464       17464            306634928          1537336690    NaN   
17465       17465            306634928          1537336690    NaN   
18844       18844            329565856          1560267618    NaN   
18847       18847            329604595          1560306357    NaN   
18848       18848            329604595          1560306357    NaN   
19149       19149            334595024          1565296786    NaN   
19150       19150            334595024          1565296786    NaN   
20262       20262            353082362          1583784124    NaN   
20265       20265            353107293          1583809055    NaN   
20266       20266            353107293          1583809055    NaN   
20568       20568            358114658          1588816420    NaN   
20569       20569            358114658          1588816420    NaN   
21763       21763            377967062          1608668824    NaN   
21765       21765            377995596          1608697358    NaN   
21766       21766            377995596          1608697358    NaN   
22068       22068            383003468          1613705230    NaN   
22069       22069            383003468          1613705230    NaN   
23927       23927            413896276          1644598038    NaN   
23930       23930            413934520          1644636282    NaN   
23931       23931            413934520          1644636282    NaN   
24232       24232            418925370          1649627132    NaN   
24235       24235            419124702          1649826464    NaN   

       Eyetracker timestamp                          Event Event value  \
0                       NaN                 RecordingStart         NaN   
221                     NaN  Eye tracker Calibration start         NaN   
15745                   NaN                     MouseEvent  Down, Left   
15753                   NaN                     MouseEvent    Up, Left   
17137                   NaN                     MouseEvent  Down, Left   
17145                   NaN                     MouseEvent    Up, Left   
17161                   NaN    Eye tracker Calibration end         NaN   
17162                   NaN             ImageStimulusStart      blank2   
17464                   NaN               ImageStimulusEnd      blank2   
17465                   NaN             ImageStimulusStart  視線誘導問題.001   
18844                   NaN                  KeyboardEvent       Space   
18847                   NaN               ImageStimulusEnd  視線誘導問題.001   
18848                   NaN             ImageStimulusStart      blank2   
19149                   NaN               ImageStimulusEnd      blank2   
19150                   NaN             ImageStimulusStart  視線誘導問題.002   
20262                   NaN                  KeyboardEvent       Space   
20265                   NaN               ImageStimulusEnd  視線誘導問題.002   
20266                   NaN             ImageStimulusStart      blank2   
20568                   NaN               ImageStimulusEnd      blank2   
20569                   NaN             ImageStimulusStart  視線誘導問題.004   
21763                   NaN                  KeyboardEvent       Space   
21765                   NaN               ImageStimulusEnd  視線誘導問題.004   
21766                   NaN             ImageStimulusStart      blank2   
22068                   NaN               ImageStimulusEnd      blank2   
22069            

In [16]:
# "視線誘導問題." を含む行を抽出
df_filtered = df_sensor_nan[df_sensor_nan["Event value"].astype(str).str.contains("視線誘導問題.")].copy()

# 結果の確認（先頭5行を表示）
df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']]
print(df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']])

       Recording timestamp  Computer timestamp Event value
17465            306634928          1537336690  視線誘導問題.001
18847            329604595          1560306357  視線誘導問題.001
19150            334595024          1565296786  視線誘導問題.002
20265            353107293          1583809055  視線誘導問題.002
20569            358114658          1588816420  視線誘導問題.004
21765            377995596          1608697358  視線誘導問題.004
22069            383003468          1613705230  視線誘導問題.003
23930            413934520          1644636282  視線誘導問題.003


### 各問題の開始時刻
|       |Recording timestamp|  Computer timestamp| Event value|
|-------|-------------------|--------------------|------------|
|17465|306634928|1537336690|視線誘導問題.001|
|18847|329604595|1560306357|視線誘導問題.001|
|19150|334595024|1565296786|視線誘導問題.002|
|20265|353107293|1583809055|視線誘導問題.002|
|20569|358114658|1588816420|視線誘導問題.004|
|21765|377995596|1608697358|視線誘導問題.004|
|22069|383003468|1613705230|視線誘導問題.003|
|23930|413934520|1644636282|視線誘導問題.003|

In [17]:
df_filtered.loc[:,['Recording timestamp', 'Computer timestamp', 'Event value']].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-7/start-stop-timing.csv")

In [18]:
df_sensor_mouse

Unnamed: 0  Recording timestamp  Computer timestamp Sensor  \
1               1                 7718          1230709480  Mouse   
3               3                23441          1230725203  Mouse   
4               4                39413          1230741175  Mouse   
5               5                55359          1230757121  Mouse   
6               6                71367          1230773129  Mouse   
...           ...                  ...                 ...    ...   
17152       17152            301432364          1532134126  Mouse   
17153       17153            301450377          1532152139  Mouse   
17154       17154            301510384          1532212146  Mouse   
17156       17156            301528368          1532230130  Mouse   
17158       17158            301544387          1532246149  Mouse   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
1                       NaN   NaN         NaN           NaN           NaN   
3                       NaN   NaN         NaN           NaN           NaN   
4                       NaN   NaN         NaN           NaN           NaN   
5                       NaN   NaN         NaN           NaN           NaN   
6                       NaN   NaN         NaN           NaN           NaN   
...                     ...   ...         ...           ...           ...   
17152                   NaN   NaN         NaN           NaN           NaN   
17153                   NaN   NaN         NaN           NaN           NaN   
17154                   NaN   NaN         NaN           NaN           NaN   
17156                   NaN   NaN         NaN           NaN           NaN   
17158                   NaN   NaN         NaN           NaN           NaN   

       Gaze point left X  ...  Original Media height  Eye movement type  \
1                    NaN  ...                    NaN                NaN   
3                    NaN  ...                    NaN       EyesNotFound   
4                    NaN  ...                    NaN       EyesNotFound   
5                    NaN  ...                    NaN       EyesNotFound   
6                    NaN  ...                    NaN       EyesNotFound   
...                  ...  ...                    ...                ...   
17152                NaN  ...                    NaN                NaN   
17153                NaN  ...                    NaN                NaN   
17154                NaN  ...                    NaN            Saccade   
17156                NaN  ...                    NaN            Saccade   
17158                NaN  ...                    NaN       EyesNotFound   

       Eye movement event duration  Eye movement type index  Fixation point X  \
1                              NaN                      NaN               NaN   
3                            204.0                      1.0               NaN   
4                            204.0                      1.0               NaN   
5                            204.0                      1.0               NaN   
6                            204.0                      1.0               NaN   
...                            ...                      ...               ...   
17152                          NaN                      NaN               NaN   
17153                          NaN                      NaN               NaN   
17154                         25.0                    252.0               NaN   
17156                         25.0                    252.0               NaN   
17158                         79.0                    109.0               NaN   

       Fixation point Y  Fixation point X (MCSnorm)  \
1                   NaN                         NaN   
3                   NaN                         NaN   
4                   NaN                         NaN   
5                   NaN                         NaN   
6                   NaN                         NaN   
...                 ...                         .

In [19]:
df_sensor_eye

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
2               2                18880          1230720642  Eye Tracker   
7               7                74350          1230776112  Eye Tracker   
10             10               105498          1230807260  Eye Tracker   
12             12               121924          1230823686  Eye Tracker   
14             14               138565          1230840327  Eye Tracker   
...           ...                  ...                 ...          ...   
24229       24229            418886911          1649588673  Eye Tracker   
24230       24230            418904247          1649606009  Eye Tracker   
24231       24231            418920291          1649622053  Eye Tracker   
24233       24233            418936881          1649638643  Eye Tracker   
24234       24234            418954171          1649655933  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
2              1.246045e+09   NaN         NaN           NaN           NaN   
7              1.246100e+09   NaN         NaN           NaN           NaN   
10             1.246131e+09   NaN         NaN           NaN           NaN   
12             1.246148e+09   NaN         NaN           NaN           NaN   
14             1.246164e+09   NaN         NaN           NaN           NaN   
...                     ...   ...         ...           ...           ...   
24229          1.664887e+09   NaN         NaN          82.0        1049.0   
24230          1.664904e+09   NaN         NaN          73.0        1065.0   
24231          1.664920e+09   NaN         NaN          56.0        1056.0   
24233          1.664937e+09   NaN         NaN          65.0        1050.0   
24234          1.664954e+09   NaN         NaN          37.0        1046.0   

       Gaze point left X  ...  Original Media height  Eye movement type  \
2                    NaN  ...                    NaN       EyesNotFound   
7                    NaN  ...                    NaN       EyesNotFound   
10                   NaN  ...                    NaN       EyesNotFound   
12                   NaN  ...                    NaN       EyesNotFound   
14                   NaN  ...                    NaN       EyesNotFound   
...                  ...  ...                    ...                ...   
24229              122.0  ...                 1080.0           Fixation   
24230              109.0  ...                 1080.0           Fixation   
24231              117.0  ...                 1080.0           Fixation   
24233              130.0  ...                    NaN           Fixation   
24234               82.0  ...                    NaN           Fixation   

       Eye movement event duration  Eye movement type index  Fixation point X  \
2                            204.0                      1.0               NaN   
7                            204.0                      1.0               NaN   
10                           204.0                      1.0               NaN   
12                           204.0                      1.0               NaN   
14                           204.0                      1.0               NaN   
...                            ...                      ...               ...   
24229                        103.0                    505.0              66.0   
24230                        103.0                    505.0              66.0   
24231                        103.0                    505.0              66.0   
24233                        103.0                    505.0              66.0   
24234                        103.0                    505.0              66.0   

       Fixation point Y  Fixation point X (MCSnorm)  \
2                   NaN                         NaN   
7                   NaN                         NaN   
10                  NaN                         NaN   
12                  NaN                         NaN   
14                  NaN         

## Mouse以外のデータを抽出

In [20]:
df_cleaned_ver2 = df_cleaned[df_cleaned['Sensor'] != "Mouse"]
nunique_series_ver2 = df_cleaned_ver2.nunique(dropna=False)
single_unique_cols_ver2 = nunique_series_ver2[nunique_series_ver2 == 1].index.tolist()
# カラム名とそのユニーク値を出力
for col in single_unique_cols_ver2:
    unique_val = df_cleaned_ver2[col].unique()
    print(f"{col}: {unique_val}")

Mouse position X: [nan]
Mouse position Y: [nan]


In [21]:
df_cleaned_ver2 = df_cleaned_ver2.drop(columns=single_unique_cols_ver2)

In [22]:
print(df_cleaned_ver2.nunique(dropna=False).to_string())

Unnamed: 0                             24037
Recording timestamp                    24028
Computer timestamp                     24028
Sensor                                     2
Eyetracker timestamp                   24008
Event                                      9
Event value                                9
Gaze point X                            1566
Gaze point Y                             924
Gaze point left X                       1552
Gaze point left Y                        953
Gaze point right X                      1560
Gaze point right Y                       864
Gaze direction left X                  10548
Gaze direction left Y                   9918
Gaze direction left Z                   4291
Gaze direction right X                 10803
Gaze direction right Y                  9808
Gaze direction right Z                  4600
Pupil diameter left                     1555
Pupil diameter right                    1496
Pupil diameter filtered                 1539
Validity l

In [23]:
df_cleaned_ver2.head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
0            0                    0          1230701762          NaN   
2            2                18880          1230720642  Eye Tracker   
7            7                74350          1230776112  Eye Tracker   
10          10               105498          1230807260  Eye Tracker   
12          12               121924          1230823686  Eye Tracker   

    Eyetracker timestamp           Event Event value  Gaze point X  \
0                    NaN  RecordingStart         NaN           NaN   
2           1.246045e+09             NaN         NaN           NaN   
7           1.246100e+09             NaN         NaN           NaN   
10          1.246131e+09             NaN         NaN           NaN   
12          1.246148e+09             NaN         NaN           NaN   

    Gaze point Y  Gaze point left X  ...  Presented Media position Y (DACSpx)  \
0            NaN                NaN  ...                                  NaN   
2            NaN                NaN  ...                                  NaN   
7            NaN                NaN  ...                                  NaN   
10           NaN                NaN  ...                                  NaN   
12           NaN                NaN  ...                                  NaN   

    Original Media width  Original Media height  Eye movement type  \
0                    NaN                    NaN                NaN   
2                    NaN                    NaN       EyesNotFound   
7                    NaN                    NaN       EyesNotFound   
10                   NaN                    NaN       EyesNotFound   
12                   NaN                    NaN       EyesNotFound   

    Eye movement event duration  Eye movement type index  Fixation point X  \
0                           NaN                      NaN               NaN   
2                         204.0                      1.0               NaN   
7                         204.0                      1.0               NaN   
10                        204.0                      1.0               NaN   
12                        204.0                      1.0               NaN   

    Fixation point Y  Fixation point X (MCSnorm)  Fixation point Y (MCSnorm)  
0                NaN                         NaN                         NaN  
2                NaN                         NaN                         NaN  
7                NaN                         NaN                         NaN  
10               NaN                         NaN                         NaN  
12               NaN                         NaN                         NaN  

[5 rows x 55 columns]

In [24]:
# 欠損値も含めてユニークな値の個数をカウント
nunique_series_ver3 = df_cleaned_ver2.nunique(dropna=False)

# ユニークな値が1つのカラム名リスト
single_unique_cols_ver3 = nunique_series_ver3[nunique_series_ver3 == 2].index.tolist()

# カラム名とそのユニーク値を出力
for col in single_unique_cols_ver3:
    unique_val = df_cleaned_ver2[col].unique()
    print(f"{col}: {unique_val}")


Sensor: [nan 'Eye Tracker']
Presented Media width: [  nan 1920.]
Presented Media height: [  nan 1080.]
Presented Media position X (DACSpx): [nan  0.]
Presented Media position Y (DACSpx): [nan  0.]
Original Media width: [  nan 1920.]
Original Media height: [  nan 1080.]


### Sensor=Mouse以外に着目した後に要素が少ない特徴量について
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Mouse position X| マウス位置 X|`nan`|
|Mouse position Y| マウス位置 Y|`nan`|
|Sensor| センサー|`nan`, `Eye Tracker`|
|Presented Media width|提示されたメディアの幅|`nan` `1920`|
|Presented Media height|提示されたメディアの高さ|`nan`, `1080`|
|Presented Media position X (DACSpx)| 提示されたメディア位置 X|`nan`,  `0`|
|Presented Media position Y (DACSpx)| 提示されたメディアの位置 Y|`nan`,  `0`|
|Original Media width| オリジナルメディアの幅|`nan`, `1920`|
|Original Media height| オリジナルメディアの高さ|`nan`,`1080`|

In [25]:
df_sensor_eye

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
2               2                18880          1230720642  Eye Tracker   
7               7                74350          1230776112  Eye Tracker   
10             10               105498          1230807260  Eye Tracker   
12             12               121924          1230823686  Eye Tracker   
14             14               138565          1230840327  Eye Tracker   
...           ...                  ...                 ...          ...   
24229       24229            418886911          1649588673  Eye Tracker   
24230       24230            418904247          1649606009  Eye Tracker   
24231       24231            418920291          1649622053  Eye Tracker   
24233       24233            418936881          1649638643  Eye Tracker   
24234       24234            418954171          1649655933  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
2              1.246045e+09   NaN         NaN           NaN           NaN   
7              1.246100e+09   NaN         NaN           NaN           NaN   
10             1.246131e+09   NaN         NaN           NaN           NaN   
12             1.246148e+09   NaN         NaN           NaN           NaN   
14             1.246164e+09   NaN         NaN           NaN           NaN   
...                     ...   ...         ...           ...           ...   
24229          1.664887e+09   NaN         NaN          82.0        1049.0   
24230          1.664904e+09   NaN         NaN          73.0        1065.0   
24231          1.664920e+09   NaN         NaN          56.0        1056.0   
24233          1.664937e+09   NaN         NaN          65.0        1050.0   
24234          1.664954e+09   NaN         NaN          37.0        1046.0   

       Gaze point left X  ...  Original Media height  Eye movement type  \
2                    NaN  ...                    NaN       EyesNotFound   
7                    NaN  ...                    NaN       EyesNotFound   
10                   NaN  ...                    NaN       EyesNotFound   
12                   NaN  ...                    NaN       EyesNotFound   
14                   NaN  ...                    NaN       EyesNotFound   
...                  ...  ...                    ...                ...   
24229              122.0  ...                 1080.0           Fixation   
24230              109.0  ...                 1080.0           Fixation   
24231              117.0  ...                 1080.0           Fixation   
24233              130.0  ...                    NaN           Fixation   
24234               82.0  ...                    NaN           Fixation   

       Eye movement event duration  Eye movement type index  Fixation point X  \
2                            204.0                      1.0               NaN   
7                            204.0                      1.0               NaN   
10                           204.0                      1.0               NaN   
12                           204.0                      1.0               NaN   
14                           204.0                      1.0               NaN   
...                            ...                      ...               ...   
24229                        103.0                    505.0              66.0   
24230                        103.0                    505.0              66.0   
24231                        103.0                    505.0              66.0   
24233                        103.0                    505.0              66.0   
24234                        103.0                    505.0              66.0   

       Fixation point Y  Fixation point X (MCSnorm)  \
2                   NaN                         NaN   
7                   NaN                         NaN   
10                  NaN                         NaN   
12                  NaN                         NaN   
14                  NaN         

## 各問題毎のデータへ切り分け

In [26]:
# 結果を辞書に格納
extracted_data = {}

targets = df_filtered["Event value"].unique()
for target in targets:
    timestamps = df_filtered[df_filtered["Event value"] == target]["Recording timestamp"].iloc[0:2]
    if len(timestamps) < 2:
        continue  # start/endが不足している場合スキップ

    start, end = timestamps.iloc[0], timestamps.iloc[1]

    extracted = df_sensor_eye[
        (df_sensor_eye["Recording timestamp"] >= start) &
        (df_sensor_eye["Recording timestamp"] <= end)
    ].copy()

    extracted_data[target] = extracted  # 日本語のキーOK

In [27]:
extracted_data["視線誘導問題.002"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
19151       19151            334600274          1565302036  Eye Tracker   
19152       19152            334616496          1565318258  Eye Tracker   
19153       19153            334633247          1565335009  Eye Tracker   
19154       19154            334650197          1565351959  Eye Tracker   
19155       19155            334666368          1565368130  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
19151          1.580605e+09   NaN         NaN         817.0         571.0   
19152          1.580622e+09   NaN         NaN         817.0         572.0   
19153          1.580638e+09   NaN         NaN         809.0         565.0   
19154          1.580655e+09   NaN         NaN         816.0         558.0   
19155          1.580671e+09   NaN         NaN         822.0         571.0   

       Gaze point left X  ...  Original Media height  Eye movement type  \
19151              829.0  ...                 1080.0           Fixation   
19152              832.0  ...                 1080.0           Fixation   
19153              824.0  ...                 1080.0           Fixation   
19154              837.0  ...                 1080.0           Fixation   
19155              831.0  ...                 1080.0           Fixation   

       Eye movement event duration  Eye movement type index  Fixation point X  \
19151                        982.0                    260.0             826.0   
19152                        982.0                    260.0             826.0   
19153                        982.0                    260.0             826.0   
19154                        982.0                    260.0             826.0   
19155                        982.0                    260.0             826.0   

       Fixation point Y  Fixation point X (MCSnorm)  \
19151             585.0                      0.4301   
19152             585.0                      0.4301   
19153             585.0                      0.4301   
19154             585.0                      0.4301   
19155             585.0                      0.4301   

       Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
19151                      0.5413               NaN               NaN  
19152                      0.5413               NaN               NaN  
19153                      0.5413               NaN               NaN  
19154                      0.5413               NaN               NaN  
19155                      0.5413               NaN               NaN  

[5 rows x 57 columns]

In [28]:
print(extracted_data["視線誘導問題.002"].nunique(dropna=False).to_string())

Unnamed: 0                             1113
Recording timestamp                    1113
Computer timestamp                     1113
Sensor                                    1
Eyetracker timestamp                   1113
Event                                     1
Event value                               1
Gaze point X                            507
Gaze point Y                            309
Gaze point left X                       518
Gaze point left Y                       367
Gaze point right X                      521
Gaze point right Y                      313
Gaze direction left X                  1017
Gaze direction left Y                  1003
Gaze direction left Z                   889
Gaze direction right X                 1046
Gaze direction right Y                 1000
Gaze direction right Z                  891
Pupil diameter left                     372
Pupil diameter right                    402
Pupil diameter filtered                 378
Validity left                   

In [29]:
extracted_data["視線誘導問題.001"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
17466       17466            306643402          1537345164  Eye Tracker   
17467       17467            306659463          1537361225  Eye Tracker   
17468       17468            306676269          1537378031  Eye Tracker   
17469       17469            306693323          1537395085  Eye Tracker   
17470       17470            306709334          1537411096  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
17466          1.552650e+09   NaN         NaN         981.0         405.0   
17467          1.552666e+09   NaN         NaN         979.0         406.0   
17468          1.552683e+09   NaN         NaN         984.0         416.0   
17469          1.552700e+09   NaN         NaN         988.0         412.0   
17470          1.552716e+09   NaN         NaN         951.0         430.0   

       Gaze point left X  ...  Original Media height  Eye movement type  \
17466              974.0  ...                 1080.0           Fixation   
17467              972.0  ...                 1080.0           Fixation   
17468              984.0  ...                 1080.0           Fixation   
17469              989.0  ...                 1080.0           Fixation   
17470                NaN  ...                 1080.0            Saccade   

       Eye movement event duration  Eye movement type index  Fixation point X  \
17466                        383.0                    187.0             979.0   
17467                        383.0                    187.0             979.0   
17468                        383.0                    187.0             979.0   
17469                        383.0                    187.0             979.0   
17470                         16.0                    265.0               NaN   

       Fixation point Y  Fixation point X (MCSnorm)  \
17466             400.0                      0.5099   
17467             400.0                      0.5099   
17468             400.0                      0.5099   
17469             400.0                      0.5099   
17470               NaN                         NaN   

       Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
17466                      0.3702               NaN               NaN  
17467                      0.3702               NaN               NaN  
17468                      0.3702               NaN               NaN  
17469                      0.3702               NaN               NaN  
17470                         NaN               NaN               NaN  

[5 rows x 57 columns]

In [30]:
extracted_data["視線誘導問題.003"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
22070       22070            383008481          1613710243  Eye Tracker   
22071       22071            383025619          1613727381  Eye Tracker   
22072       22072            383041639          1613743401  Eye Tracker   
22073       22073            383058399          1613760161  Eye Tracker   
22074       22074            383075542          1613777304  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
22070          1.629011e+09   NaN         NaN        1261.0         238.0   
22071          1.629028e+09   NaN         NaN        1263.0         235.0   
22072          1.629044e+09   NaN         NaN        1262.0         236.0   
22073          1.629060e+09   NaN         NaN        1266.0         236.0   
22074          1.629078e+09   NaN         NaN        1267.0         233.0   

       Gaze point left X  ...  Original Media height  Eye movement type  \
22070             1281.0  ...                 1080.0           Fixation   
22071             1280.0  ...                 1080.0           Fixation   
22072             1278.0  ...                 1080.0           Fixation   
22073             1279.0  ...                 1080.0           Fixation   
22074             1281.0  ...                 1080.0           Fixation   

       Eye movement event duration  Eye movement type index  Fixation point X  \
22070                        799.0                    389.0            1269.0   
22071                        799.0                    389.0            1269.0   
22072                        799.0                    389.0            1269.0   
22073                        799.0                    389.0            1269.0   
22074                        799.0                    389.0            1269.0   

       Fixation point Y  Fixation point X (MCSnorm)  \
22070             228.0                      0.6607   
22071             228.0                      0.6607   
22072             228.0                      0.6607   
22073             228.0                      0.6607   
22074             228.0                      0.6607   

       Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
22070                      0.2114               NaN               NaN  
22071                      0.2114               NaN               NaN  
22072                      0.2114               NaN               NaN  
22073                      0.2114               NaN               NaN  
22074                      0.2114               NaN               NaN  

[5 rows x 57 columns]

In [31]:
extracted_data["視線誘導問題.004"].head()

Unnamed: 0  Recording timestamp  Computer timestamp       Sensor  \
20570       20570            358130157          1588831919  Eye Tracker   
20571       20571            358146945          1588848707  Eye Tracker   
20572       20572            358163927          1588865689  Eye Tracker   
20573       20573            358180029          1588881791  Eye Tracker   
20574       20574            358196864          1588898626  Eye Tracker   

       Eyetracker timestamp Event Event value  Gaze point X  Gaze point Y  \
20570          1.604134e+09   NaN         NaN         997.0         181.0   
20571          1.604151e+09   NaN         NaN         996.0         175.0   
20572          1.604168e+09   NaN         NaN         996.0         173.0   
20573          1.604184e+09   NaN         NaN        1000.0         176.0   
20574          1.604200e+09   NaN         NaN        1000.0         173.0   

       Gaze point left X  ...  Original Media height  Eye movement type  \
20570             1006.0  ...                 1080.0           Fixation   
20571             1004.0  ...                 1080.0           Fixation   
20572             1007.0  ...                 1080.0           Fixation   
20573             1015.0  ...                 1080.0           Fixation   
20574             1018.0  ...                 1080.0           Fixation   

       Eye movement event duration  Eye movement type index  Fixation point X  \
20570                       1065.0                    316.0             997.0   
20571                       1065.0                    316.0             997.0   
20572                       1065.0                    316.0             997.0   
20573                       1065.0                    316.0             997.0   
20574                       1065.0                    316.0             997.0   

       Fixation point Y  Fixation point X (MCSnorm)  \
20570             173.0                      0.5194   
20571             173.0                      0.5194   
20572             173.0                      0.5194   
20573             173.0                      0.5194   
20574             173.0                      0.5194   

       Fixation point Y (MCSnorm)  Mouse position X  Mouse position Y  
20570                      0.1604               NaN               NaN  
20571                      0.1604               NaN               NaN  
20572                      0.1604               NaN               NaN  
20573                      0.1604               NaN               NaN  
20574                      0.1604               NaN               NaN  

[5 rows x 57 columns]

In [32]:
for key in extracted_data.keys():
    print(f"Key: {key}")
    df = extracted_data[key]
    nunique_series = df.nunique(dropna=False)
    single_unique_cols = nunique_series[nunique_series == 1].index.tolist()
    for col in single_unique_cols:
        unique_val = df[col].unique()
        print(f"  {col}: {unique_val}")

Key: 視線誘導問題.001
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.001']
  Presented Media name: ['視線誘導問題.001.jpeg']
  Presented Media width: [1920.]
  Presented Media height: [1080.]
  Presented Media position X (DACSpx): [0.]
  Presented Media position Y (DACSpx): [0.]
  Original Media width: [1920.]
  Original Media height: [1080.]
  Mouse position X: [nan]
  Mouse position Y: [nan]
Key: 視線誘導問題.002
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.002']
  Presented Media name: ['視線誘導問題.002.jpeg']
  Presented Media width: [1920.]
  Presented Media height: [1080.]
  Presented Media position X (DACSpx): [0.]
  Presented Media position Y (DACSpx): [0.]
  Original Media width: [1920.]
  Original Media height: [1080.]
  Mouse position X: [nan]
  Mouse position Y: [nan]
Key: 視線誘導問題.004
  Sensor: ['Eye Tracker']
  Event: [nan]
  Event value: [nan]
  Presented Stimulus name: ['視線誘導問題.004']
  Presen

In [33]:
# 空の辞書に結果を格納
cleaned_data = {}

for key, df in extracted_data.items():
    nunique_series = df.nunique(dropna=False)
    single_unique_cols = nunique_series[nunique_series == 1].index.tolist()

    # そのカラムを除外して新しいデータフレームを作成
    df_cleaned = df.drop(columns=single_unique_cols).copy()

    cleaned_data[key] = df_cleaned

    print(f"{key}: Removed {len(single_unique_cols)} columns -> Remaining {df_cleaned.shape[1]} columns")


視線誘導問題.001: Removed 13 columns -> Remaining 44 columns
視線誘導問題.002: Removed 13 columns -> Remaining 44 columns
視線誘導問題.004: Removed 13 columns -> Remaining 44 columns
視線誘導問題.003: Removed 13 columns -> Remaining 44 columns


### カラムの要素が1つしかない変数の説明 写真1
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-7`|
|Recording name| 録画名|`Recording3`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:49:27.713`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:49:27.713`|
|Recording duration| 録画時間|`419125`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`20.8`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.77`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`72`pixels|
|Average validation accuracy | 平均バリデーション精度|`17.2`mm|
|Average validation precision SD | 平均バリデーション予測SD|`2.7`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`2.5`mm|
|Average validation accuracy |平均バリデーション精度| `1.55`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.24`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.23`degrees|
|Average validation accuracy |平均バリデーション精度| `60`pixels|
|Average validation precision SD |平均バリデーション予測SD| `9`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `9`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.001`|
|Presented Media name|提示されたメディア名| `視線誘導問題.001.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|

In [34]:
cleaned_data["視線誘導問題.001"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
17466       17466            306643402          1537345164   
17467       17467            306659463          1537361225   
17468       17468            306676269          1537378031   
17469       17469            306693323          1537395085   
17470       17470            306709334          1537411096   
...           ...                  ...                 ...   
18841       18841            329524197          1560225959   
18842       18842            329540995          1560242757   
18843       18843            329558050          1560259812   
18845       18845            329574070          1560275832   
18846       18846            329590913          1560292675   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
17466          1.552650e+09         981.0         405.0              974.0   
17467          1.552666e+09         979.0         406.0              972.0   
17468          1.552683e+09         984.0         416.0              984.0   
17469          1.552700e+09         988.0         412.0              989.0   
17470          1.552716e+09         951.0         430.0                NaN   
...                     ...           ...           ...                ...   
18841          1.575530e+09        1077.0         435.0             1062.0   
18842          1.575546e+09        1079.0         434.0             1062.0   
18843          1.575563e+09        1090.0         438.0             1082.0   
18845          1.575579e+09        1089.0         443.0             1084.0   
18846          1.575596e+09        1090.0         444.0             1083.0   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
17466              397.0               988.0               413.0  ...   
17467              402.0               986.0               411.0  ...   
17468              409.0               985.0               422.0  ...   
17469              404.0               987.0               419.0  ...   
17470                NaN               951.0               430.0  ...   
...                  ...                 ...                 ...  ...   
18841              423.0              1093.0               447.0  ...   
18842              420.0              1095.0               448.0  ...   
18843              421.0              1098.0               455.0  ...   
18845              414.0              1095.0               471.0  ...   
18846              413.0              1098.0               475.0  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
17466                       0.3673                        0.5145   
17467                       0.3720                        0.5135   
17468                       0.3791                        0.5131   
17469                       0.3741                        0.5139   
17470                          NaN                        0.4954   
...                            ...                           ...   
18841                       0.3913                        0.5692   
18842                       0.3893                        0.5705   
18843                       0.3899                        0.5719   
18845                       0.3835                        0.5704   
18846                       0.3822                        0.5719   

       Gaze point right Y (MCSnorm)  Eye movement type  \
17466                        0.3827           Fixation   
17467                        0.3806           Fixation   
17468                        0.3909           Fixation   
17469                        0.3882           Fixation   
17470                        0.3980            Saccade   
...                             ...                ...   
18841                        0.4141           Fixation   
18842                        0.4151           Fixation   
18843                        0.4209           Fixation   
18845                        0.4361           Fixat

In [35]:
cleaned_data["視線誘導問題.001"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-7/user-7-pic-1.csv")

### カラムの要素が1つしかない変数の説明 写真4
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-7`|
|Recording name| 録画名|`Recording3`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:49:27.713`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:49:27.713`|
|Recording duration| 録画時間|`419125`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`20.8`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.77`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`72`pixels|
|Average validation accuracy | 平均バリデーション精度|`17.2`mm|
|Average validation precision SD | 平均バリデーション予測SD|`2.7`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`2.5`mm|
|Average validation accuracy |平均バリデーション精度| `1.55`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.24`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.23`degrees|
|Average validation accuracy |平均バリデーション精度| `60`pixels|
|Average validation precision SD |平均バリデーション予測SD| `9`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `9`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.004`|
|Presented Media name|提示されたメディア名| `視線誘導問題.004.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|


In [36]:
cleaned_data["視線誘導問題.004"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
20570       20570            358130157          1588831919   
20571       20571            358146945          1588848707   
20572       20572            358163927          1588865689   
20573       20573            358180029          1588881791   
20574       20574            358196864          1588898626   
...           ...                  ...                 ...   
21759       21759            377916511          1608618273   
21760       21760            377933469          1608635231   
21761       21761            377949725          1608651487   
21762       21762            377966430          1608668192   
21764       21764            377983393          1608685155   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
20570          1.604134e+09         997.0         181.0             1006.0   
20571          1.604151e+09         996.0         175.0             1004.0   
20572          1.604168e+09         996.0         173.0             1007.0   
20573          1.604184e+09        1000.0         176.0             1015.0   
20574          1.604200e+09        1000.0         173.0             1018.0   
...                     ...           ...           ...                ...   
21759          1.623919e+09        1159.0         527.0             1140.0   
21760          1.623936e+09        1147.0         532.0             1126.0   
21761          1.623952e+09        1149.0         535.0             1137.0   
21762          1.623969e+09        1157.0         531.0             1142.0   
21764          1.623986e+09        1149.0         539.0             1142.0   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
20570              144.0               988.0               219.0  ...   
20571              134.0               987.0               216.0  ...   
20572              135.0               986.0               211.0  ...   
20573              136.0               985.0               215.0  ...   
20574              136.0               983.0               210.0  ...   
...                  ...                 ...                 ...  ...   
21759              539.0              1177.0               515.0  ...   
21760              537.0              1168.0               527.0  ...   
21761              552.0              1161.0               519.0  ...   
21762              547.0              1172.0               516.0  ...   
21764              545.0              1156.0               533.0  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
20570                       0.1332                        0.5144   
20571                       0.1243                        0.5142   
20572                       0.1246                        0.5136   
20573                       0.1261                        0.5130   
20574                       0.1257                        0.5119   
...                            ...                           ...   
21759                       0.4988                        0.6131   
21760                       0.4976                        0.6084   
21761                       0.5110                        0.6045   
21762                       0.5060                        0.6104   
21764                       0.5048                        0.6018   

       Gaze point right Y (MCSnorm)  Eye movement type  \
20570                        0.2028           Fixation   
20571                        0.1995           Fixation   
20572                        0.1955           Fixation   
20573                        0.1989           Fixation   
20574                        0.1947           Fixation   
...                             ...                ...   
21759                        0.4773           Fixation   
21760                        0.4882           Fixation   
21761                        0.4802           Fixation   
21762                        0.4777           Fixat

In [37]:
cleaned_data["視線誘導問題.004"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-7/user-7-pic-4.csv")

### カラムの要素が1つしかない変数の説明 写真2
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-7`|
|Recording name| 録画名|`Recording3`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:49:27.713`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:49:27.713`|
|Recording duration| 録画時間|`419125`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`20.8`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.77`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`72`pixels|
|Average validation accuracy | 平均バリデーション精度|`17.2`mm|
|Average validation precision SD | 平均バリデーション予測SD|`2.7`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`2.5`mm|
|Average validation accuracy |平均バリデーション精度| `1.55`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.24`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.23`degrees|
|Average validation accuracy |平均バリデーション精度| `60`pixels|
|Average validation precision SD |平均バリデーション予測SD| `9`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `9`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.002`|
|Presented Media name|提示されたメディア名| `視線誘導問題.002.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|

In [38]:
cleaned_data["視線誘導問題.002"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
19151       19151            334600274          1565302036   
19152       19152            334616496          1565318258   
19153       19153            334633247          1565335009   
19154       19154            334650197          1565351959   
19155       19155            334666368          1565368130   
...           ...                  ...                 ...   
20259       20259            353038050          1583739812   
20260       20260            353054779          1583756541   
20261       20261            353071779          1583773541   
20263       20263            353087923          1583789685   
20264       20264            353104697          1583806459   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
19151          1.580605e+09         817.0         571.0              829.0   
19152          1.580622e+09         817.0         572.0              832.0   
19153          1.580638e+09         809.0         565.0              824.0   
19154          1.580655e+09         816.0         558.0              837.0   
19155          1.580671e+09         822.0         571.0              831.0   
...                     ...           ...           ...                ...   
20259          1.599042e+09         846.0         257.0              841.0   
20260          1.599059e+09         848.0         253.0              841.0   
20261          1.599076e+09         848.0         251.0              834.0   
20263          1.599092e+09         847.0         256.0              836.0   
20264          1.599109e+09         847.0         252.0              839.0   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
19151              596.0               805.0               547.0  ...   
19152              590.0               803.0               554.0  ...   
19153              591.0               795.0               538.0  ...   
19154              585.0               795.0               531.0  ...   
19155              595.0               812.0               547.0  ...   
...                  ...                 ...                 ...  ...   
20259              236.0               850.0               277.0  ...   
20260              234.0               856.0               271.0  ...   
20261              233.0               861.0               270.0  ...   
20263              239.0               857.0               272.0  ...   
20264              238.0               855.0               266.0  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
19151                       0.5518                        0.4192   
19152                       0.5465                        0.4181   
19153                       0.5476                        0.4141   
19154                       0.5416                        0.4140   
19155                       0.5512                        0.4231   
...                            ...                           ...   
20259                       0.2182                        0.4428   
20260                       0.2166                        0.4457   
20261                       0.2155                        0.4486   
20263                       0.2216                        0.4466   
20264                       0.2200                        0.4451   

       Gaze point right Y (MCSnorm)  Eye movement type  \
19151                        0.5062           Fixation   
19152                        0.5125           Fixation   
19153                        0.4978           Fixation   
19154                        0.4916           Fixation   
19155                        0.5066           Fixation   
...                             ...                ...   
20259                        0.2569           Fixation   
20260                        0.2511           Fixation   
20261                        0.2501           Fixation   
20263                        0.2520           Fixat

In [39]:
cleaned_data["視線誘導問題.002"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-7/user-7-pic-2.csv")

### カラムの要素が1つしかない変数の説明 写真3
|変数名(英語)|変数名(日本語)|中身|
|------------|--------------|----|
|Project name|プロジェクト名|`JD4`|
|Export date|エクスポート日| `2025/07/18`|
|Participant name| 参加者名|`user-7`|
|Recording name| 録画名|`Recording3`|
|Recording date|録画日(JSC) |`2025/07/18`|
|Recording date UTC| 録画日(UTC)|`2025/07/18`|
|Recording start time|録画開始時刻(JSC) |`16:49:27.713`|
|Recording start time UTC| 録画開始時刻(UTC) |`07:49:27.713`|
|Recording duration| 録画時間|`419125`ミリ秒|
|Timeline name| タイムライン名|`Timeline1`|
|Recording Fixation filter name| 録画の注視フィルタ名|`Tobii I-VT (Fixation)`|
|Recording software version| 録画ソフトのバージョン|`24.21.435`|
|Recording resolution height|録画解像度(高さ) |`1080`px|
|Recording resolution width|録画解像度(幅) | `1920`px|
|Recording monitor latency| モニター遅延|`10`ミリ秒|
|Average calibration accuracy | 平均キャリブレーション精度|`20.8`mm|
|Average calibration accuracy | 平均キャリブレーション精度|`1.77`degrees|
|Average calibration accuracy | 平均キャリブレーション精度|`72`pixels|
|Average validation accuracy | 平均バリデーション精度|`17.2`mm|
|Average validation precision SD | 平均バリデーション予測SD|`2.7`mm|
|Average validation precision RMS |平均バリデーション予測 RMS |`2.5`mm|
|Average validation accuracy |平均バリデーション精度| `1.55`degrees|
|Average validation precision SD |平均バリデーション予測 SD| `0.24`degrees|
|Average validation precision RMS |平均バリデーション予測 RMS| `0.23`degrees|
|Average validation accuracy |平均バリデーション精度| `60`pixels|
|Average validation precision SD |平均バリデーション予測SD| `9`pixels|
|Average validation precision RMS |平均バリデーション予測 RMS| `9`pixels|
|---------------------------------|--------------------------|-----------|
|  Sensor|センサー| `Eye Tracker`|
|  Event|イベント |`nan`|
|Event value|イベントの中身| `nan`|
|Presented Stimulus name|提示された刺激名| `視線誘導問題.003`|
|Presented Media name|提示されたメディア名| `視線誘導問題.003.jpeg`|
|Presented Media width|提示されたメディア幅|`1920`px|
|Presented Media height|提示されたメディア高さ|`1080`px|
|Presented Media position X (DACSpx)|提示されたメディア位置 X|`0`|
|Presented Media position Y (DACSpx)|提示されたメディア位置 Y|`0`|
|Original Media width|オリジナルメディアの幅|`1920`px|
|Original Media height |オリジナルメディアの高さ|`1080`px|
|Mouse position X|マウス位置 X|`nan`|
|Mouse position Y|マウス位置 Y|  `nan`|


In [40]:
cleaned_data["視線誘導問題.003"]

Unnamed: 0  Recording timestamp  Computer timestamp  \
22070       22070            383008481          1613710243   
22071       22071            383025619          1613727381   
22072       22072            383041639          1613743401   
22073       22073            383058399          1613760161   
22074       22074            383075542          1613777304   
...           ...                  ...                 ...   
23924       23924            413861469          1644563231   
23925       23925            413878059          1644579821   
23926       23926            413894687          1644596449   
23928       23928            413911484          1644613246   
23929       23929            413927983          1644629745   

       Eyetracker timestamp  Gaze point X  Gaze point Y  Gaze point left X  \
22070          1.629011e+09        1261.0         238.0             1281.0   
22071          1.629028e+09        1263.0         235.0             1280.0   
22072          1.629044e+09        1262.0         236.0             1278.0   
22073          1.629060e+09        1266.0         236.0             1279.0   
22074          1.629078e+09        1267.0         233.0             1281.0   
...                     ...           ...           ...                ...   
23924          1.659862e+09        1006.0         752.0              980.0   
23925          1.659878e+09        1000.0         747.0              978.0   
23926          1.659895e+09        1008.0         744.0              998.0   
23928          1.659912e+09        1008.0         760.0              991.0   
23929          1.659928e+09        1010.0         746.0              979.0   

       Gaze point left Y  Gaze point right X  Gaze point right Y  ...  \
22070              200.0              1242.0               276.0  ...   
22071              198.0              1245.0               272.0  ...   
22072              203.0              1245.0               268.0  ...   
22073              199.0              1253.0               272.0  ...   
22074              193.0              1253.0               273.0  ...   
...                  ...                 ...                 ...  ...   
23924              777.0              1032.0               727.0  ...   
23925              776.0              1022.0               718.0  ...   
23926              783.0              1019.0               705.0  ...   
23928              781.0              1025.0               739.0  ...   
23929              776.0              1042.0               716.0  ...   

       Gaze point left Y (MCSnorm)  Gaze point right X (MCSnorm)  \
22070                       0.1847                        0.6466   
22071                       0.1833                        0.6486   
22072                       0.1880                        0.6486   
22073                       0.1843                        0.6523   
22074                       0.1787                        0.6527   
...                            ...                           ...   
23924                       0.7198                        0.5375   
23925                       0.7187                        0.5321   
23926                       0.7249                        0.5307   
23928                       0.7232                        0.5340   
23929                       0.7183                        0.5426   

       Gaze point right Y (MCSnorm)  Eye movement type  \
22070                        0.2553           Fixation   
22071                        0.2520           Fixation   
22072                        0.2484           Fixation   
22073                        0.2519           Fixation   
22074                        0.2526           Fixation   
...                             ...                ...   
23924                        0.6735           Fixation   
23925                        0.6645           Fixation   
23926                        0.6525           Fixation   
23928                        0.6839           Fixat

In [41]:
cleaned_data["視線誘導問題.003"].to_csv("/content/drive/MyDrive/class/情報デザイン論/加工データ csv/user-7/user-7-pic-3.csv")